sound/soc/bcm/allo-piano-dac-plus.c

```diff
@@ -102,7 +104,7 @@ static int __snd_allo_piano_dsp_program(struct snd_soc_pcm_runtime *rtd,
 
 	if ((lowpass > 14) || (lowpass < 0))
 		lowpass = 3;
-	if ((mode > 2) || (mode < 0))
+	if ((mode > 4) || (mode < 0))
```

Comment:
```mode >= ARRAY_SIZE(allo_piano_mode_texts)```
mode can't be < 0 as it is an unsigned int. I'm surprised that doesn't throw a warning.